driver-sync/src/main/com/mongodb/client/internal/ChangeStreamIterableImpl.java

```diff
@@ -135,13 +147,38 @@
         return this;
     }
 
+    @Override
+    public MongoCursor<ChangeStreamDocument<TResult>> iterator() {
+        return cursor();
+    }
+
+    @Override
+    public MongoChangeStreamCursor<ChangeStreamDocument<TResult>> cursor() {
+        return new MongoChangeStreamCursorImpl<ChangeStreamDocument<TResult>>(execute(), codec, initialResumeToken());
+    }
+
+    @Nullable
+    @Override
+    public ChangeStreamDocument<TResult> first() {
+        MongoChangeStreamCursor<ChangeStreamDocument<TResult>> cursor = cursor();
+        try {
+            if (!cursor.hasNext()) {
+                return null;
+            }
+            return cursor.next();
+        } finally {
+            cursor.close();
+        }
+    }
+
     @Override
     public ReadOperation<BatchCursor<ChangeStreamDocument<TResult>>> asReadOperation() {
-        return createChangeStreamOperation(codec);
+        throw new UnsupportedOperationException();
```

Comment:
The same applies here - if a user calls `map` on the change stream iterable does it work or throw?

Reply:
As mentioned above, I wrote a test using `map` that succeeds.